### PR TITLE
280 - QA: Anchor View samples button to samples section on experiment page

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number
Closing #280

(This branch was created based on PR https://github.com/AlexsLemonade/refinebio-web/pull/301.)

## Purpose/Implementation Notes
Added the auto-scroll animation to the samples table section, when a user clicks the "View Samples" button to access the experiment page via the search page. The UI scrolls the samples table section to the top of the page as the data is loaded.

Screenshot 2024-01-03 at 7 33 35 PM

Please see the latest UI [here](https://refinebio-k72hn2i5o-ccdl.vercel.app/search).

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by dvenprasad — APPROVED** [review]
Looks good! 
🦓 🦓 🦓

File: src/contexts/RefinebioContext.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ export const RefinebioContextProvider = ({ children }) => {
     'requested-experiments',
     []
   )
+  const [samplesTableData, setSamplesTableData] = useState({})
```
Comment (davidsmejia) [inline src/contexts/RefinebioContext.js:25]: I'm having trouble understanding why we should put this data here. This is our global application context and should be used for data that needs to be persisted and accessed in multiple locations.

**Review by davidsmejia — COMMENTED** [review]
Left a question about approach. Do you think that `scrollView` would be better served written as a hook?

File: src/contexts/RefinebioContext.js (hunk 1 above)
Comment (nozomione) [reply src/contexts/RefinebioContext.js]: The `samplesTableData` was previously shared with `src/pages/experiments/[accession_code]/[experiment]/index` to calculate the correct scroll position of the samples table scroll animation (ensuring the table data was fully loaded). But it has been moved from the global context and is defined in `SamplesTableContext` (at cee5282ac17f02756443223a717fb6cb5a07cdd5).

Instead, I am now using the [ResizeObserver API](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) to handle the calculation for the samples table position on layout changes (at f5edd0278fd6caed0e1b947db07e8f9b6e8b3c26).

I've also made a small revision to `helpers/scrollTop` (previously was named `scrollToTop`) and adjusted the codebase (at ec71236b7c3577d370dff6239eb7e0f9f4aa2f2f). Thank you, David!

**Review by davidsmejia — COMMENTED** [review]
Can we first clean up the renaming of context hook state getters/setters.

**Comment by nozomione** [comment]
I've updated this PR and it's ready for another look. Thank you David!

**Comment by nozomione** [comment]
As discussed in 1:1, I've replaced the `querySelector` method with the Context API.

It now handles access to the `<header />`  element using `LayoutRefsContext` and its hook `useLayoutRefs`  (dde79d12fa284fee0a3a7d5d4a301f6c8b80a38c). Additionally, I've cleaned up the `sampleTablesData` states.

Also I am going to create a separate issue for further refactoring of `useSamplesTableManager` for the next sprint, as we discussed 👍 

This PR is ready for your review, thank you David!

**Pull request merged**